// data-analysis-python\src\get_stock_info\get_stock_info.proto

syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/get_stock_info";

// 株式の全フィールドを取得するためのサービスを定義
service GetStockInfoService {
  rpc GetStockInfo (GetStockInfoRequest) returns (GetStockInfoResponse);
}

// リクエストメッセージ（国名を指定）
message GetStockInfoRequest {
  string country = 1;  // 国名
}

// 株式情報を格納するメッセージ
message StockInfo {
  string country = 1;          // 国名
  string symbol = 2;           // 証券コード
  string name = 3;             // 銘柄名
  string full_name = 4;        // 正式名称
  string isin = 5;             // ISINコード
  string currency = 6;         // 通貨
  string stock_exchange = 7;   // 証券取引所
  string sector = 8;           // セクター
  string industry = 9;         // 業界
}

// 株式情報を格納するレスポンスメッセージ
message GetStockInfoResponse {
  repeated StockInfo stocks = 1;  // 株式情報のリスト
}


// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_stock_info --python_out=src/get_stock_info --grpc_python_out=src/get_stock_info src/get_stock_info/get_stock_info.proto

// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_stock_info --go_out=./src/service/ms_gateway/get_stock_info --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_stock_info --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_stock_info/get_stock_info.proto
